src/nvim/diff.c

```diff
@@ -1777,6 +1777,9 @@ bool diff_linematch(diff_T *dp)
   int tsize = 0;
   for (int i = 0; i < DB_COUNT; i++) {
     if ( curtab->tp_diffbuf[i] != NULL ) {
+      if (dp->df_count[i] < 0) {
+        return 0;
+      }
```

Comment:
I just checked in this change.

dp->df_count[i] determines the size of a diff hunk in buffer i

I think this may be a rare bug. In some instance it is possible for the
dp->df_count[i] to be a negative number.

I verified that this is unrelated to changes that I've made for the linematch
feature.

you can reproduce it with the files and changes included here on this branch

https://github.com/jwhite510/neovim/commits/bug2

by running
make && ./build/bin/nvim -u NONE -S bug.vim

a block of memory in the diff alignment algorithm is determined by multiplying
together the length of the diff block on each buffer, and when one of those
lengths (dp->df_count) is negative, it uses a negative number to allocate a
diff block, and that negative number is cast to a size_t, and that ends up
being a huge number, it tries to allocate a huge block of memory and it can't
and it crashes.

So, I just disabled running the algorithm when this is the case.

I've not yet taken the time to investigate how the number became negative

Reply:
Good to know. Make sure to comment the code accordingly.

Reply:
commented